### PR TITLE
build: deprecate tools/bazel.rc

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

*Description*:
*Risk Level*: Low
*Testing*: CI
*Docs Changes*:
*Release Notes*:

### CODE REVIEW

**Comment by htuch** [comment]
Do we need changes in envoy-filter-example? I'm trying to figure out how best to message this, since we break folks who might be symlinking such as Lyft. @junr03 does this require changes in your setup?

**Comment by mattklein123** [comment]
Yes this will break people that are symlinking. What is the right way to handle for all of these people? Just import from a root .bazelrc?

**Comment by lizan** [comment]
With latest commit, if you have `tools/bazel.rc` symlinked you will get error like:
```
INFO: Reading rc options for 'build' from /home/lizan/github/lizan/envoy/build/tools/bazel.rc:
  Inherited 'common' options: --dummy_unknown_option=ERROR: tools/bazel.rc is being processed, either due to old version of bazel or wrongly symlinked from .bazelrc.
ERROR: Unrecognized option: --dummy_unknown_option=ERROR: tools/bazel.rc is being processed, either due to old version of bazel or wrongly symlinked from .bazelrc.
INFO: Invocation ID: 50637508-5174-4674-aa55-a82045d5c4ab
```

This shouldn't affect other build process if it is not symlinked or old bazel processing that file.

File: tools/bazel.rc [hunk 1]
```diff
@@ -1,53 +1,4 @@
...
-build:clang-msan --define ENVOY_CONFIG_MSAN=1
-build:clang-msan --copt -fsanitize=memory
-build:clang-msan --linkopt -fsanitize=memory
-build:clang-msan --define tcmalloc=disabled
-build:clang-msan --copt -fsanitize-memory-track-origins=2
-build:clang-msan --linkopt -fuse-ld=lld
-
-# Test options
-test --test_env=HEAPCHECK=normal --test_env=PPROF_PATH
+common --dummy_unknown_option="ERROR: tools/bazel.rc is being processed, either due to old version of bazel or wrongly symlinked from .bazelrc."
```
Comment (htuch) [inline tools/bazel.rc]: Maybe provide a helpful remedy? People don't like being told they are wrong :)

**Pull request merged**